firmware/lib/spectrum_scanner.rb

```
@@ -0,0 +1,20 @@
+  def initialize(receiver, start_hz:, step_hz:, count:, sleeper: ->(_ms) {}, wait_ms: 0)
+    @receiver = receiver
+    @start_hz = start_hz
+    @step_hz  = step_hz
+    @count    = count
+    @sleeper  = sleeper
+    @wait_ms  = wait_ms
+  end
+
+  def scan
+    @count.times do |i|
+      freq = @start_hz + @step_hz * i
```

RuboCop (1.86.1)

[warning] Wrap expressions with varying precedence with parentheses to avoid ambiguity.

(Lint/AmbiguousOperatorPrecedence)